services/postprocessing/pkg/service/service.go

```diff
@@ -65,12 +66,13 @@ func NewPostprocessingService(ctx context.Context, stream events.Stream, logger
 
 // Run to fulfil Runner interface
 func (pps *PostprocessingService) Run() error {
-	// Spawn workers that'll concurrently work the queue
+	wg := sync.WaitGroup{}
+
 	for i := 0; i < pps.c.Workers; i++ {
-		go (func() {
+		wg.Add(1)
+		go func() {
```

Comment:
```suggestion
		go func() {
		    wg.Add(1)
		    defer wg.Done()
```

You need to call `wg.Done()` otherwise this will hang forever

Reply:
updated